src/modules/silentpayments/bench_impl.h

```diff
@@ -0,0 +1,168 @@
+    CHECK(secp256k1_ec_pubkey_parse(data->ctx, &data->spend_pubkey, spend_pubkey, pubkeylen));
+    memcpy(data->scan_key, scan_key, 32);
+    memcpy(data->smallest_outpoint, smallest_outpoint, 36);
+}
+
+static void bench_silentpayments_output_scan(void* arg, int iters) {
+    int i, k = 0;
+    bench_silentpayments_data *data = (bench_silentpayments_data*)arg;
+    secp256k1_silentpayments_prevouts_summary prevouts_summary;
+
+    for (i = 0; i < iters; i++) {
+        unsigned char shared_secret[33];
+        secp256k1_xonly_pubkey xonly_output;
+        CHECK(secp256k1_silentpayments_recipient_prevouts_summary_parse(data->ctx, &prevouts_summary, data->input_pubkey33));
+        CHECK(secp256k1_silentpayments_recipient_create_shared_secret(data->ctx,
+            shared_secret,
+            data->scan_key,
+            &prevouts_summary
+        ));
+        CHECK(secp256k1_silentpayments_recipient_create_output_pubkey(data->ctx,
+            &xonly_output,
+            shared_secret,
+            &data->spend_pubkey,
+            k
+        ));
+    }
+}
+
+static void bench_silentpayments_full_tx_scan(void* arg, int iters, int use_labels) {
+    int i;
+    size_t n_found = 0;
+    secp256k1_silentpayments_found_output *found_output_ptrs[2];
+    const secp256k1_xonly_pubkey *tx_output_ptrs[2];
+    const secp256k1_xonly_pubkey *tx_input_ptrs[2];
+    bench_silentpayments_data *data = (bench_silentpayments_data*)arg;
+    secp256k1_silentpayments_prevouts_summary prevouts_summary;
+    const secp256k1_silentpayments_label_lookup label_lookup_fn = use_labels ? label_lookup : NULL;
+    const void *label_context = use_labels ? label_cache : NULL;
+
+    for (i = 0; i < 2; i++) {
+        found_output_ptrs[i] = &data->found_outputs[i];
+        tx_output_ptrs[i] = &data->tx_outputs[i];
+        tx_input_ptrs[i] = &data->tx_inputs[i];
+    }
+    for (i = 0; i < iters; i++) {
+        CHECK(secp256k1_silentpayments_recipient_prevouts_summary_create(data->ctx,
+            &prevouts_summary,
+            data->smallest_outpoint,
+            tx_input_ptrs, 2,
+            NULL, 0
+        ));
+        CHECK(secp256k1_silentpayments_recipient_scan_outputs(data->ctx,
+            found_output_ptrs, &n_found,
+            tx_output_ptrs, 2,
+            data->scan_key,
+            &prevouts_summary,
+            &data->spend_pubkey,
+            label_lookup_fn, label_context)
+        );
+        CHECK(n_found == 0);
+    }
+}
+
+static void bench_silentpayments_full_scan(void *arg, int iters) {
+    bench_silentpayments_full_tx_scan(arg, iters, 0);
+}
+
+static void bench_silentpayments_full_scan_with_labels(void *arg, int iters) {
+    bench_silentpayments_full_tx_scan(arg, iters, 1);
+}
+
+static void run_silentpayments_bench(int iters, int argc, char** argv) {
+    bench_silentpayments_data data;
+    int d = argc == 1;
+
+    /* create a context with no capabilities */
```

Comment:
nit: the context has indeed capabilities, so could just remove the comment (or s/no capabilities/signing capabilities/, as done in the EllSwift benchmarks)

Reply:
I opted to remove the comment. It didn't feel like it was adding any new information, beyond just describing what the code is doing.